### ISSUE TITLE
Insert after v3 upgrade.

### ISSUE THREAD

**Issue by sarabu07** [linq2db/linq2db#2504]
We have a table in oracle with following create statement. COLUMN_C value is get from a sequence.
```sql
-- TABLE_A definition

CREATE TABLE "TABLE_A" 
   (	
    "COLUMN_A" NUMBER(20,0) NOT NULL ENABLE NOVALIDATE, 
	"COLUMN_B" NUMBER(6,0) NOT NULL ENABLE NOVALIDATE, 
	"COLUMN_C" NUMBER(6,0) NOT NULL ENABLE NOVALIDATE, 
	"COLUMN_D" VARCHAR2(10) NOT NULL ENABLE, 
	"COLUMN_E" VARCHAR2(10) NOT NULL ENABLE, 
	"COLUMN_F" NUMBER(20,0), 
	"COLUMN_G" NUMBER(20,0) NOT NULL ENABLE, 
	"COLUMN_H" DATE, 
	"COLUMN_I" VARCHAR2(10) NOT NULL ENABLE, 
	"COLUMN_J" VARCHAR2(10) NOT NULL ENABLE
	 CONSTRAINT "PK_TABLE_A" PRIMARY KEY ("COLUMN_A", "COLUMN_B", "COLUMN_C")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 27222016 NEXT 52428800 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "WO_INDEX"  ENABLE, 
	 SUPPLEMENTAL LOG DATA (PRIMARY KEY) COLUMNS
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 60 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING
  STORAGE(INITIAL 409600000 NEXT 52428800 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT) ;
```
when we insert a record into table till version 2.9.8 it was working after v3 upgrade it was failing for error message
ORA-01400: cannot insert NULL into ("PROD"."TABLE_A"."COLUMN_C"). 
Could you help us if the support needs to be added in v3 or we need to change the insert statement?
The Table class is as follows:

```cs
   [Table(Name = "TABLE_A")]
    public sealed class TABLEA
    {
        [PrimaryKey]
        [Column(Name = "COLUMN_A"), NotNull]
        public long COLUMNA { get; set; }

      
        [PrimaryKey]
        [Column(Name = "COLUMN_B"), NotNull]
        public int COLUMNB { get; set; }

       
        [PrimaryKey]
        [Column(Name = "COLUMN_C"), NotNull, SequenceName(ProviderName.Oracle, "seq_A")]
        public int COLUMNC { get; set; }
}
//column and table name is changed.
```
we are using InsertWithInt64IdentityAsync to insert into the table.

**Comment by sdanyliv** [comment]
Will check what happened. Anyway try add `[Identity]` attribute maybe it will be workaround.

**Comment by sarabu07** [comment]
@sdanyliv I tried using below code and still not working.

        
        [PrimaryKey]
        [Column(Name = "COLUMN_C"), NotNull, Identity,SequenceName(ProviderName.Oracle, "seq_A")]
        public int COLUMNC { get; set; }

**Comment by viceroypenguin** [comment]
Have you tried adding `SkipOnInsert = true` to the `Column` attribute as follows:
```csharp
    [PrimaryKey]
    [Column(Name = "COLUMN_C", SkipOnInsert = true), NotNull, Identity, SequenceName(ProviderName.Oracle, "seq_A")]
    public int COLUMNC { get; set; }
```

**Comment by sarabu07** [comment]
@viceroypenguin after applying above changes to code also its not working.

**Comment by viceroypenguin** [comment]
@sarabu07  I will be doing some testing on this today.

**Comment by MaceWindu** [comment]
Issue confirmed. As workaround remove first parameter from `SequenceName(ProviderName.Oracle, "seq_A")` -> `SequenceName("seq_A")`

**Comment by viceroypenguin** [comment]
@sarabu07 We have released version 3.1.4 to fix this.

**Comment by sarabu07** [comment]
The issue is resolved now. I tested in my app with version 3.1.5
